### DISCUSSION TITLE
When should we use inline=True in jax.jit?

### DISCUSSION THREAD

**Discussion by nasyxx** [jax-ml/jax#14832] [Q&A]
According to my knowledge, we should make the function inline as much as possible for efficiency. But is it still correct in JAX?

I have some questions about this. For example, I have the following two functions:

```
def f1(x):
     return x + 1

def f2(x):
     return f1(x) + 1
```

Is it the same between inline jit `f1`, then jit `f2`, and only jit `f2`?

**Comment by hawkinsp** [comment]
The only effect `inline=True` will have is making the IR smaller if you have a trivial function, really. We added it mostly so our internal `jit`ted functions like `+` don't gum up the IR with many little function calls.

Given our current compiler (XLA) eventually inlines everything anyway, I would basically ignore it for the moment.

**Reply by soraros** [reply]
I've seen latest development in XLA like [#1728](https://github.com/openxla/xla/pull/1728). Does this mean `inline` can do something more interesting in the future?
